src/encoded/static/components/filegallery.js

```diff
@@ -1223,41 +1236,103 @@ DatasetFiles.defaultProps = {
 };
 
 
-// File display widget, showing a facet list, a table, and a graph (and maybe a BioDalliance).
-// This component only triggers the data retrieval, which is done with a search for files associated
-// with the given experiment (in this.props.context). An odd thing is we specify query-string parameters
-// to the experiment URL, but they apply to the file search -- not the experiment itself.
-export const FileGallery = ({ context, hideGraph, hideControls, collapseNone, altFilterDefault, showReplicateNumber }, reactContext) => (
-    <FetchedData>
-        <Param name="data" url={`/search/?limit=all&type=File&dataset=${context['@id']}`} />
-        <Param name="schemas" url="/profiles/" />
+/**
+ * File display widget, showing a facet list, Valis, a table, and a graph. This component only
+ * triggers the data retrieval, which is done with a search for files associated with the given
+ * experiment (in `context`). The source of files gets specified either by `fileQuery` or
+ * `fileProp`. With `fileQuery` a search request with the given query string goes to the server,
+ * and displayed files come from the results. With `fileProp` a property embedded in the dataset
+ */
+export const FileGallery = ({
+    context,
+    fileQuery,
+    files,
+    analyses,
+    hideGraph,
+    hideControls,
+    collapseNone,
+    showReplicateNumber,
+    showDetailedTracks,
+}, reactContext) => {
+    // Holds all schemas loaded from a request; used by the file graph.
+    const [schemas, setSchemas] = React.useState(null);
+    // Holds file search results from a request.
+    const [data, setData] = React.useState([]);
```

Comment:
name does not seem to match the functionality. Use of variable name "data" is used for general multi-purpose variables/constants